### PR TITLE
8265019 : Update tests for additional TestNG test permissions

### PR DESCRIPTION
**Pull request by LanceAndersen**
Hi all,

Please review the following patch which adds additional permissions needed for when JTREG upgrades to a newer version of TestNG.

Best,
Lance

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265019](https://bugs.openjdk.java.net/browse/JDK-8265019): Update tests for additional TestNG test permissions

### Reviewers
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to 20ef2bd07d3980537d28c0a307beffa067a75db0
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to 20ef2bd07d3980537d28c0a307beffa067a75db0
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to 20ef2bd07d3980537d28c0a307beffa067a75db0

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3471/head:pull/3471` \
`$ git checkout pull/3471`

Update a local copy of the PR: \
`$ git checkout pull/3471` \
`$ git pull https://git.openjdk.java.net/jdk pull/3471/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3471`

View PR using the GUI difftool: \
`$ git pr show -t 3471`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3471.diff">https://git.openjdk.java.net/jdk/pull/3471.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/lang/ProcessHandle/PermissionTest.java [hunk 1]
```diff
@@ -207,6 +208,11 @@ public void setBasicPermissions() {
         permissions.add(new RuntimePermission("createSecurityManager"));
         permissions.add(new PropertyPermission("testng.show.stack.frames",
                 "read"));
+        permissions.add(new PropertyPermission("testng.thread.affinity", "read"));
+        permissions.add(new PropertyPermission("testng.mode.dryrun", "read"));
+        permissions.add(new PropertyPermission("testng.report.xml.name", "read"));
+        permissions.add(new ReflectPermission("suppressAccessChecks"));
+        permissions.add(new PropertyPermission("testng.timezone", "read"));
```
Comment (AlanBateman) [inline test/jdk/java/lang/ProcessHandle/PermissionTest.java:215]: might be better to group the testng.* properties so they aren't mixed up with the other runtime permissions.

**Review by naotoj — APPROVED** [review]
Looks good.

**Review by bplb — APPROVED** [review]

File: test/jdk/java/sql/testng/util/TestPolicy.java [hunk 2]
```diff
@@ -23,14 +23,8 @@
...
-import java.security.AllPermission;
-import java.security.CodeSource;
-import java.security.Permission;
-import java.security.PermissionCollection;
-import java.security.Permissions;
-import java.security.Policy;
-import java.security.ProtectionDomain;
-import java.security.SecurityPermission;
+import java.lang.reflect.ReflectPermission;
+import java.security.*;
```
Comment (AlanBateman) [inline test/jdk/java/sql/testng/util/TestPolicy.java:27]: My guess is that you didn't mean the change the import.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**